src/tokens/saga.ts

```diff
@@ -221,7 +229,22 @@ export function* fetchTokenBalancesSaga() {
           .toFixed()
       }
     }
-    yield* put(setTokenBalances(tokens))
+
+    const importedTokensWithBalance = showImportedTokens
+      ? yield* call(
+          fetchImportedTokenBalances,
+          address as Address,
+          importedTokens,
+          fetchedBalancesByTokenId
+        )
```

Comment:
I'm slightly confused, we're fetching balances for imported tokens also passing a variable saying balances are already fetched since it's called `fetchBalancesByTokenId`?

oh ok it seems it's in case we already have the balance in the known set.

Reply:
Yes, sorry, the logic is a bit weird. But the backend can return the balance for some tokens, so to avoid fetching the balance again I decided this approach.

Maybe I overthought this, and fetching the balance again was just fine and the code would be clearer.

Reply:
yeah it took me a while to understand this too (essentially this happens because blockscout supports some tokens that we do not support, right?) - can you please add a comment to avoid future head scratching?

Reply:
Yes, I'll add a comment